scripts/close-issue-channel.js

```diff
@@ -33,130 +34,131 @@ startBot(async (client) => {
+    );
+    if (archiveChannel === undefined) {
+      console.log(`No archive channel found.`);
+      return false;
+    }
+    const oldChannels = [...channels.cache.values()].filter(
+      (ch) =>
+        ch.parentId === archiveChannel.parentId && ch.name.includes(`website-issue-${issueNumber}`)
+    );
+    for (const oldChannel of oldChannels) {
+      const timestamp = formatRFC882PST(new Date());
+      const title = `╭
 │    __/__/_ 
 │   __/__/_  ${oldChannel.name}
 │    /  /
 │
 ╰─ archived ${timestamp}`;
-			const footer = `╭${'─'.repeat(timestamp.length + 11)}╮
+      const footer = `╭${'─'.repeat(timestamp.length + 11)}╮
 │ archived ${timestamp} │
 ╰${'─'.repeat(timestamp.length + 11)}╯`;
-			const allMessages = await fetchAllMessages(oldChannel);
-			const fileContent =
-				title + '\n\n' + allMessages.map(formatMessage).join('\n') + '\n\n' + footer;
-			if (dev) {
-				console.log(fileContent);
-				return true;
-			}
-			const attachment = Buffer.from(fileContent);
-			const message = await archiveChannel.send({
-				files: [{ name: `${oldChannel.name}.txt`, attachment }],
-			});
-			await message.pin();
-			await oldChannel.delete();
-		}
-		success = true;
-	} catch (error) {
-		console.log(error);
-	}
-	return success;
+      const allMessages = await fetchAllMessages(oldChannel);
+      const fileContent =
+        title + '\n\n' + allMessages.map(formatMessage).join('\n') + '\n\n' + footer;
+      if (dev) {
+        console.log(fileContent);
+        return true;
+      }
+      const attachment = Buffer.from(fileContent);
+      const message = await archiveChannel.send({
+        files: [{ name: `${oldChannel.name}.txt`, attachment }],
+      });
+      await message.pin();
+      await oldChannel.delete();
+    }
+    success = true;
+  } catch (error) {
+    console.log(error);
+  }
+  return success;
 };
 
 const fetchAllMessages = async (channel, limit = 2e3) => {
-	const result = [];
-	const softLimit = 100;
-	let lastId;
-	while (true) {
-		const options = { limit: softLimit };
-		if (lastId) options.before = lastId;
-		const messages = await channel.messages.fetch(options);
-		if (messages === undefined) break;
-		result.push(...messages.values());
-		lastId = messages.last().id;
-		if (messages.size != softLimit || result >= limit) break;
-		await sleep(250); // Let's not hammer the API.
-	}
-	return result.sort((a, b) => a.createdAt.getTime() - b.createdAt.getTime());
+  const result = [];
+  const softLimit = 100;
+  let lastId;
+  // eslint-disable-next-line no-constant-condition
+  while (true) {
+    const options = { limit: softLimit };
+    if (lastId) options.before = lastId;
+    const messages = await channel.messages.fetch(options);
+    if (messages === undefined) break;
+    result.push(...messages.values());
+    lastId = messages.last().id;
+    if (messages.size != softLimit || result >= limit) break;
+    await sleep(250); // Let's not hammer the API.
```

Comment:
You don't have to worry about this. Discord.js will do it for you if it needs
to.

Reply:
Agh, I forgot to check this in the previous PR. Maybe it can be a follow-up PR.

Reply:
Resolved by <https://github.com/EthanThatOneKid/acmcsuf.com/pull/199/commits/d5ea2afb3fbbcb1abaec6ed71e5773c6fb08dd36>.